src/repo.rs

```diff
@@ -1685,4 +1812,42 @@ impl Repo {
             false
         }
     }
+
+    fn enqueue_share_decisions<'a, I: Iterator<Item = &'a RepoId>>(
```

Comment:
I would rather see:

```
fn enqueue_share_decisions(
    doc_info: &mut DocumentInfo,
    share_authorizations_to_poll: &mut HashSet<RepoId>,
    /// the same below
)
```

Which would only process repos for the document which do not have already an authorization.